### PR TITLE
Add printing_merge_request_link_enabled attribute to projects

### PR DESCRIPTION
**Pull request by marceloboeira**
As mentioned on #782, the field is available on GitLab but still not
possible to use on terraform

Reference:
- https://github.com/gitlabhq/terraform-provider-gitlab/issues/782

### CODE REVIEW

**Comment by marceloboeira** [comment]
This requires https://github.com/xanzy/go-gitlab/pull/1343 to be merged and released, together with an update of the go-gitlab client.

**Comment by armsnyder** [comment]
Thanks for the PR @marceloboeira. Once that go-gitlab feature is merged, you can feel free to upgrade it as part of this PR. Could you also add that field to the test?

**Comment by marceloboeira** [comment]
> 

just added to the default of the test, I guess most tests aren't field-specific, but we can add one if that makes sense.

**Comment by armsnyder** [comment]
Yep, that works. Most of our tests set all the fields, and the in the subsequent "upgrade" test step they change all the fields.

**Comment by marceloboeira** [comment]
UPDATE 24.01.22

The PR on gitlab client side has been merged, but still need a release so that we can use a "final" release rather than a commit sha.

Once the release is done I can update the code to point to that release and move it forward.

**Comment by timofurrer** [comment]
@marceloboeira there has been a go-gitlab release some days ago, which we've already integrated. Therefore, I think you could continue here.

Please make also sure to rebase before continuing as we've changed quite some things ;)

**Comment by marceloboeira** [comment]
thanks @timofurrer, I have rebased/adjusted accordingly hehe hopefully this time is a go

File: internal/provider/data_source_gitlab_project.go [hunk 1]
```diff
@@ -134,6 +134,11 @@ var _ = registerDataSource("gitlab_project", func() *schema.Resource {
 				Type:        schema.TypeBool,
 				Computed:    true,
 			},
+			"printing_merge_request_link_enabled": {
+				Description: " Show link to create/view merge request when pushing from the command line",
```
Comment (timofurrer) [inline internal/provider/data_source_gitlab_project.go:138]:
```suggestion
				Description: "Show link to create/view merge request when pushing from the command line",
```

File: internal/provider/resource_gitlab_project.go [hunk 2]
```diff
@@ -204,6 +204,12 @@ var resourceGitLabProjectSchema = map[string]*schema.Schema{
 		Type:        schema.TypeBool,
 		Optional:    true,
 	},
+	"printing_merge_request_link_enabled": {
+		Description: " Show link to create/view merge request when pushing from the command line",
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project.go:208]:
```suggestion
		Description: "Show link to create/view merge request when pushing from the command line",
```

File: internal/provider/data_source_gitlab_project.go [hunk 3]
```diff
@@ -134,6 +134,11 @@ var _ = registerDataSource("gitlab_project", func() *schema.Resource {
...
+				Type:        schema.TypeBool,
+				Optional:    true,
```
Comment (timofurrer) [inline internal/provider/data_source_gitlab_project.go:140]:
```suggestion
				Computed:    true,
```

**Review by timofurrer — CHANGES_REQUESTED** [review]

File: internal/provider/data_source_gitlab_project.go (hunk 1 above)
Comment (marceloboeira) [reply internal/provider/data_source_gitlab_project.go:138]: sorry, didn't notice those.

**Comment by armsnyder** [comment]
@marceloboeira As a tip of you didn't know, you can run `make reviewable` to make sure automated checks will pass.

**Review by timofurrer — APPROVED** [review]
lgtm 🎉

**Comment by marceloboeira** [comment]
@timofurrer I'm going to figure out the specs failing, I believe it's something related to defaults...

**Comment by marceloboeira** [comment]
> @marceloboeira As a tip if you didn't know, you can run `make reviewable` to make sure automated checks will pass.

somehow that fails, I have to review my gosetup on a new laptop :D

**Comment by armsnyder** [comment]
> somehow that fails, I have to review my gosetup on a new laptop :D

Could you let us know if you find out, so we can fix or add it to the CONTRIBUTING.md 🙏

**Comment by marceloboeira** [comment]
@timofurrer @armsnyder finally looking good ehhe

File: docs/data-sources/project.md [hunk 4]
```diff
@@ -43,6 +43,7 @@ data "gitlab_project" "example" {
 - **namespace_id** (Number) The namespace (group or user) of the project. Defaults to your user.
```
[the remark this answers is not in the document]
Comment (marceloboeira) [reply docs/data-sources/project.md:43]: they are not removed...

[the remark this answers is not in the document]
Comment (timofurrer) [reply docs/data-sources/project.md:43]: I'm sorry, that seems to be visualization bug in the GitHub mobile up, where they are rendered in some kind of red ^^

Nevermind then, ... 🎉

**Review by timofurrer — APPROVED** [review]
LGTM 🎉

**Review by armsnyder — APPROVED** [review]

**Pull request merged**